Source/player.cpp

```diff
@@ -2701,6 +2702,16 @@ void FixPlrWalkTags(const Player &player)
 			dPlayer[searchTile.x][searchTile.y] = 0;
 		}
 	}
+
+#ifdef _DEBUG
+	// Checks that FixPlrWalkTags really removes player from all dPlayer tiles
+	// FixPlrWalkTags could fail if player.position.old is not updated correctly
+	for (int y = 0; y < MAXDUNY; y++) {
+		for (int x = 0; x < MAXDUNX; x++) {
+			assert(PlayerAtPosition({ x, y }) != &player);
+		}
+	}

```

Comment:
This might indeed be very helpful for catching any further cases. Good thinking :+1: